language/documentation/book/src/abilities.md

```diff
@@ -0,0 +1,243 @@
+* `signer` has `drop`
+    * Cannot be copied and cannot be put into global storage
+* `vector<T>` may have `copy`, `drop`, and `store` depending on the abilities of `T`.
+    * See [Conditional Abilities and Generic Types](#conditional-abilities-and-generic-types) for more details.
+* Immutable references `&` and mutable references `&mut` both have `copy` and `drop`.
+    * This refers to copying and dropping the reference itself, not what they refer to.
+    * References cannot appear in global storage, hence they do not have `store`.
+
+None of the primitive types have `key`, meaning none of them can be used directly with the [global storage operations](./global-storage-operators.md).
+
+## Annotating Structs
+
+To declare that a `struct` has an ability, it is declared with `has <ability>` after the struct name but before the fields. For example:
+
+```move
+struct Ignorable has drop { f: u64 }
+struct Pair has copy, drop, store { x: u64, y: u64 }
+```
+
+In this case: `Ignorable` has the `drop` ability. `Pair` has `copy`, `drop`, and `store`.
+
+
+All of these abilities have strong guarantees over these gated operations. The operation can be performed on the value only if it has that ability; even if the value is deeply nested inside of some other collection!
+
+As such: when declaring a struct’s abilities, certain requirements are placed on the fields. All fields must satisfy these constraints. These rules are necessary so that structs satisfy the reachability rules for abilities given above. If a struct is declared with the ability...
+
+* `copy`, all fields must have `copy`
+* `drop`, all fields must have `drop`
+* `store`, all fields must have `store`
+* `key`, all fields must have `store`
+    * `key` is the only ability currently that doesn’t require itself.
+
+For example:
+
+```move
+// A struct without any abilities
+struct NoAbilities {}
+
+struct WantsCopy has copy {
+    f: NoAbilities, // ERROR 'NoAbilities' does not have 'copy'
+}
+```
+
+and similarly:
+
+```move
+// A struct without any abilities
+struct NoAbilities {}
+
+struct MyResource has key {
+    f: NoAbilities, // Error 'NoAbilities' does not have 'store'
+}
+```
+
+## Conditional Abilities and Generic Types
+
+When abilities are annotated on a generic type, not all instances of that type are guaranteed to have that ability. Consider this struct declaration:
+
+```
+struct Cup<T> has copy, drop, store, key { item: T }
+```
+
+It might be very helpful if `Cup` could hold any type, regardless of it's abilities. The type system can *see* the type parameter so it should be able to remove abilities from `Cup` if it *sees* a type parameter that would violate the guarantees for that ability.
+
+This behavior might sound a bit confusing at first, but it might be more understandable if we think about collection types. Consider `vector`, we could consider it to have the following type declaration:
```

Comment:
```suggestion
This behavior might sound a bit confusing at first, but it might be more understandable if we think about collection types. Consider `vector`, with the following type declaration:
```

Reply:
This doesn't quite work since I am trying to stress that this isn't the real declaration of the type. But I will play with it

Reply:
Well, at least fix the punctuation: you've got two full sentences separated with a comma....

Reply:
Where? Was this for a different block?

Reply:
You already fixed it.